File: Data Structure/Hash Map/Easy/LET_no350_Intersection of Two Arrays II.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> record;
        for (int i : nums1)
            record[i]++;
        
        vector<int> res;
        for (int i : nums2) {
            if (record[i]-- > 0)
                res.push_back(i);
        }
        return res;
    }
};

File: Data Structure/Hash/Easy/LET_no350_Intersection of Two Arrays II.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        m, n = len(nums1), len(nums2)
        if m > n:
            return self.intersect(nums2, nums1)
        
        memo = collections.Counter(nums1)
        res = []
        for num in nums2:
            if memo[num] > 0:
                res.append(num)
                memo[num] -= 1
        
        return res